mkosi

```diff
@@ -2492,6 +2492,28 @@ def set_root_password(args: CommandLineArguments, root: str, do_run_build_script
             patch_file(os.path.join(root, 'etc/shadow'), jj)
 
 
+def run_prepare_script(args: CommandLineArguments, root: str, do_run_build_script: bool, cached: bool) -> None:
+    if args.prepare_script is None:
+        return
+    if cached:
```

Comment:
I agree, but maybe we should do this treewide instead of here particularly, since this is basically just a copy of `run_postinst_script` (and one could maybe factor the common stuff out). I think there is some value in keeping it having the same overall structure as long as it is a mostly verbatim copy.